### DISCUSSION TITLE
CRUD accessing the API from within your extension

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#13617] [Archive]
In Directus docs, we have an [example for reading the collections list](https://docs.directus.io/extensions/modules/#example-accessing-the-api-from-within-your-extension) which works fine.

```
		this.api.get('/collections?limit=-1').then((res) => {
			this.collections = res.data.data;
		});
```

How to amend this example to:

- Create an item in a specific collection
- Read the item
- Update it
- Delete it

**Comment by br41nslug** [comment]
The Vue injected helper is a basic wrapper around the regular API where the authentication is taken care of by the core app logic.
https://docs.directus.io/reference/items

Some quick examples for each CRUD operation would be:
- [create] `this.api.post('/items/:collection', {...obj...})` https://docs.directus.io/reference/items/#create-an-item
- [read] `this.api.get('/items/:collection')` https://docs.directus.io/reference/items/#get-items
- [update] `this.api.patch('/items/:collection/:id', {...obj...})` https://docs.directus.io/reference/items/#update-an-item
- [delete] `this.api.delete('/items/:collection/:id')` https://docs.directus.io/reference/items/#delete-an-item

**Comment by Pachat** [comment]
api.get works fine (`network` is a custom field in `directus_users`)
```
api.get('/users/me?fields[]=network').then((res) => {
      props.value = res.data.data.network
})
```

Now, I'd like to patch the value for the current user

```
      let another_network = {
        "network": "Another Network"
      }
      api.patch('/users/me', modif)
```

But I get
>{"errors":[{"message":"Invalid user credentials.","extensions":{"code":"INVALID_CREDENTIALS"}}]}

In my `.env`, I have 
```
REFRESH_TOKEN_TTL="7d"
REFRESH_TOKEN_COOKIE_SECURE=false
REFRESH_TOKEN_COOKIE_SAME_SITE="lax"
REFRESH_TOKEN_COOKIE_NAME="direct_55_refresh_token"

```
but I don't know how to allow the api.patch. Could you help?
